File: promotion_service/PromotionService.Core/Domain/Dto/Response/Resource.cs
using PromotionService.Core.Domain.Constant;

namespace PromotionService.Core.Domain.Dto.Response
{
    public class Resource<T>
    {
        public MetaResource Meta { get; set; }
        public T Data { get; set; }

        public static Resource<T> WithMeta(MetaResource meta)
        {
            return new Resource<T>
            {
                Meta = meta
            };
        }

        public static Resource<T> Success(T data)
        {
            return new Resource<T>
            {
                Meta = new MetaResource(1000, "Success"),
                Data = data
            };
        }
    }
}

File: rating_service/src/main/java/huy/project/rating_service/ui/resource/Resource.java
package huy.project.rating_service.ui.resource;

import lombok.*;
import org.springframework.http.HttpStatus;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Resource<T> {
    private MetaResource meta;
    private T data;

    public Resource(T data) {
        this.meta = new MetaResource(HttpStatus.OK.value(), "Success");
        this.data = data;
    }

    public Resource(Integer code, String message){
        this.meta = new MetaResource(code, message);
        this.data = null;
    }
}